### ISSUE TITLE
Avoidable NaNs on Chaikin Money Flow (CMF)

### ISSUE THREAD

**Issue by lluissalord** [twopirllc/pandas-ta#37]
Hi,

Using your (amazing) library, I have found that CMF is producing a lot of NaNs on my data. Checking which was the reason I have figured out that in my data there are a lot of cases where 'high' and 'low' price are exactly the same, hence it produces NaNs due to this part of the code:

https://github.com/twopirllc/pandas-ta/blob/a1dac5922f544af795fd6311df02406fc2728ebe/pandas_ta/volume/cmf.py#L22-L23

Hence my proposal is to add an epsilon component
`from sys import float_info as sflt`

`ad *= volume / (hl_range + sflt.epsilon)`

What is your opinion on that?

Thank you for all your work,
Lluis

**Comment by lluissalord** [comment]
Same behaviour on the following codes:

https://github.com/twopirllc/pandas-ta/blob/a1dac5922f544af795fd6311df02406fc2728ebe/pandas_ta/momentum/bop.py#L15-L16

https://github.com/twopirllc/pandas-ta/blob/a1dac5922f544af795fd6311df02406fc2728ebe/pandas_ta/volume/ad.py#L20-L21

https://github.com/twopirllc/pandas-ta/blob/a1dac5922f544af795fd6311df02406fc2728ebe/pandas_ta/overlap/kama.py#L23-L24

**Comment by twopirllc** [comment]
Hello @lluissalord,

I am glad you find the module helpful.  Thanks for finding issues that I have not noticed.
What symbol or asset are you using that has high=low?  Have you added ```sflt.epsilon``` to your code base and tested it out? I would be interested in your results in the interim.

I will take a look at it as soon as I can.

Thanks,
Kevin

**Comment by lluissalord** [comment]
Hi @twopirllc,

I am using it on bitcoin, but on historical data with a frequency of minutes. Probably there are some minutes where there is no trading (most probably on the early starts) or is due to data cleaning that has been done previously on the historical data (not by me at least).

I have added the `sflt.epsilon` on the code and it is working as expected, at least at the samples where I looked in detail seems okay because in the moments where high=low, it was also close=open (which is kind of logical), hence the result of the indicator was 0.
